### PR TITLE
mil use case

### PR DESCRIPTION
**Pull request by unly**
Use-Case demo using the Malfunction Indicator Light of the w3c VIS standard.

Signed-off-by: Leon Graser <leon.graser@bosch-si.com>

### CODE REVIEW

File: examples/malfunction-indicator-light/influxdb-rest/src/main/java/org/eclipse/kuksa/influxdb/rest/HttpQuery.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/epl-2.0/index.php
+ *
+ * *****************************************************************************
+ */
+package org.eclipse.kuksa.influxdb.rest;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
```
Comment (rohoet) [inline examples/malfunction-indicator-light/influxdb-rest/src/main/java/org/eclipse/kuksa/influxdb/rest/HttpQuery.java]: This is third party and needs to be checked, right?

File: examples/malfunction-indicator-light/mail-notification/src/main/java/org/eclipse/kuksa/notification/mail/smtp/SmtpSender.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/epl-2.0/index.php
+ *
+ * *****************************************************************************
+ */
+package org.eclipse.kuksa.notification.mail.smtp;
+
+import org.eclipse.kuksa.notification.mail.MailSender;
+import org.eclipse.kuksa.notification.mail.entity.Mail;
+import org.simplejavamail.email.Email;
```
Comment (rohoet) [inline examples/malfunction-indicator-light/mail-notification/src/main/java/org/eclipse/kuksa/notification/mail/smtp/SmtpSender.java]: Also third party content that needs to be checked I guess.

File: examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/epl-2.0/index.php
+ *
+ * *****************************************************************************
+ */
+package org.eclipse.kuksa.mil.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
```
Comment (rohoet) [inline examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java]: Third party content.

File: examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
```
Comment (rohoet) [inline examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java]: Third party content.

File: examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.google.maps.GeoApiContext;
```
Comment (rohoet) [inline examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java]: Third party content.

File: examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java (hunk 4 above)
Comment (rohoet) [reply examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java]: Done (piggy backed existing CQ for `Jackson Databind 2.9.7`)

File: examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java (hunk 3 above)
Comment (rohoet) [reply examples/malfunction-indicator-light/mil-service/src/main/java/org/eclipse/kuksa/mil/service/Oberserver.java]: Done (piggy backed existing CQ for `Jackson Core 2.9.7`)

**Comment by rohoet** [comment]
Thanks for the PR!
Please help me with my comments. Could you please provide source URL, project URL, and version of the third party content?

**Comment by unly** [comment]
Hey @rohoet yes this PR includes several third party dependencies. You can find them in the respective gradle build files.
- https://github.com/eclipse/kuksa.cloud/pull/50/files#diff-5a1fc0dcef11b81619db03e98def9adf
- https://github.com/eclipse/kuksa.cloud/pull/50/files#diff-b01ceb981af9b14c38d4a86e9fddfefc
- https://github.com/eclipse/kuksa.cloud/pull/50/files#diff-d61deb2e0e80a169697b6148edbec388

Keep in mind that those dependencies require further, transitive dependencies. Personally I can't file CQs with the Eclipse Foundation.

File: examples/malfunction-indicator-light/influxdb-rest/src/main/java/org/eclipse/kuksa/influxdb/rest/HttpQuery.java (hunk 1 above)
Comment (rohoet) [reply examples/malfunction-indicator-light/influxdb-rest/src/main/java/org/eclipse/kuksa/influxdb/rest/HttpQuery.java]: CQ started

**Comment by unly** [comment]
@eriksven i replaced the `javax.mail` in the build

File: examples/malfunction-indicator-light/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    ├── README.md
+    └── settings.gradle
+````
+
+This use-case contains three projects and their respective deployments:
+
+1. *influxdb-rest*: a REST wrapper for an InfluxDB instance
+2. *mail-notification*: a REST service to send emails from
+3. *mil-service*: sample application that checks for a possible malfunction in the car
+4. *deploy*: deployment templates for kubernetes
```
Comment (eriksven) [inline examples/malfunction-indicator-light/README.md]: in the future we should move the the 'deploy' folder to the 'deployment' folder structure

**Pull request merged**